test/object-store/realm.cpp

```diff
@@ -4173,3 +4176,165 @@ TEST_CASE("KeyPathMapping generation") {
         REQUIRE(q.count() == 0);
     }
 }
+
+#if REALM_HAVE_UV
+namespace realm::util {
+class UvLoopScheduler final : public util::Scheduler {
+public:
+    UvLoopScheduler()
+        : m_loop(std::make_unique<uv_loop_t>())
+        , m_handle(std::make_unique<uv_async_t>())
+    {
+        uv_loop_init(m_loop.get());
+        int err = uv_async_init(m_loop.get(), m_handle.get(), [](uv_async_t* handle) {
+            if (!handle->data) {
+                return;
+            }
+            auto& data = *static_cast<Data*>(handle->data);
+            if (data.close_requested) {
+                uv_close(reinterpret_cast<uv_handle_t*>(handle), [](uv_handle_t* handle) {
+                    delete reinterpret_cast<Data*>(handle->data);
+                    delete reinterpret_cast<uv_async_t*>(handle);
+                });
+            }
+            else {
+                data.queue.invoke_all();
+            }
+        });
+        if (err < 0) {
+            throw std::runtime_error(util::format("uv_async_init failed: %1", uv_strerror(err)));
+        }
+        m_handle->data = new Data;
+    }
+
+    ~UvLoopScheduler()
+    {
+        if (m_handle && m_handle->data) {
+            static_cast<Data*>(m_handle->data)->close_requested = true;
+            uv_async_send(m_handle.get());
+            // Don't delete anything here as we need to delete it from within the event loop instead
+            m_handle.release();
+        }
+    }
+
+    bool is_on_thread() const noexcept override
+    {
+        return m_id == std::this_thread::get_id();
+    }
+    bool is_same_as(const Scheduler* other) const noexcept override
+    {
+        auto o = dynamic_cast<const UvLoopScheduler*>(other);
+        return (o && (o->m_id == m_id));
+    }
+    bool can_invoke() const noexcept override
+    {
+        return true;
+    }
+
+    void invoke(util::UniqueFunction<void()>&& fn) override
+    {
+        auto& data = *static_cast<Data*>(m_handle->data);
+        data.queue.push(std::move(fn));
+        uv_async_send(m_handle.get());
+    }
+    template <class F>
+    void idle(void* data, F fn)
+    {
+        idle_handle.data = data;
+        uv_idle_init(m_loop.get(), &idle_handle);
+        uv_idle_start(&idle_handle, fn);
+    }
+    void run()
+    {
+        uv_run(m_loop.get(), UV_RUN_DEFAULT);
+    }
+    void stop()
+    {
+        uv_stop(m_loop.get());
+    }
+
+private:
+    struct Data {
+        InvocationQueue queue;
+        std::atomic<bool> close_requested = {false};
+    };
+    std::unique_ptr<uv_loop_t> m_loop;
+    std::unique_ptr<uv_async_t> m_handle;
+    uv_idle_t idle_handle;
+    std::thread::id m_id = std::this_thread::get_id();
+};
+} // namespace realm::util
+
+#if 0
```

Comment:
Don't forget to reenable it. Also would be great to shape it a bit to test some specific scenario like 'concurrent async insert and delete', so that the test wouldn't be just left as a specific reproducer.

Reply:
Has been enabled